Does not work with yarn workspaces

### Describe the bug

When using in a project that is a [yarn workspace](https://classic.yarnpkg.com/lang/en/docs/workspaces/) (on a monorepo setup), the plugin fails to find modules because of node_modules being hoisted up to the root folder:

```
yarn storybook
yarn run v1.22.17
$ start-storybook -p 6006 -s public
info @storybook/react v6.4.18
info 
(node:61699) DeprecationWarning: --static-dir CLI flag is deprecated, see:

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated---static-dir-cli-flag
(Use `node --trace-deprecation ...` to show where the warning was created)
info => Loading presets
info => Serving static files from ./public at /
info => Using implicit CSS loaders
ERR! Error: Cannot find module '<project-root>/storybook-addon-next/examples/with-tailwindcss/node_modules/next/package.json'
ERR! Require stack:
ERR! - <project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/utils.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/preset.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/core-common/dist/cjs/presets.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/core-common/dist/cjs/index.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/core-server/dist/cjs/index.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/core/dist/cjs/server.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/core/server.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/react/dist/cjs/server/index.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/react/bin/index.js
ERR!     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
ERR!     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
ERR!     at Module.require (node:internal/modules/cjs/loader:1005:19)
ERR!     at require (node:internal/modules/cjs/helpers:102:18)
ERR!     at getNextjsVersion (<project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/utils.js:9:1)
ERR!     at configureStaticImageImport (<project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/images/webpack.js:14:50)
ERR!     at configureImages (<project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/images/webpack.js:8:5)
ERR!     at <project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/preset.js:29:35
ERR!     at Generator.next (<anonymous>)
ERR!     at fulfilled (<project-root>/storybook-addon-next/examples/node_modules/tslib/tslib.js:114:62)
ERR!  Error: Cannot find module '<project-root>/storybook-addon-next/examples/with-tailwindcss/node_modules/next/package.json'
ERR! Require stack:
ERR! - <project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/utils.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/preset.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/core-common/dist/cjs/presets.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/core-common/dist/cjs/index.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/core-server/dist/cjs/index.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/core/dist/cjs/server.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/core/server.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/react/dist/cjs/server/index.js
ERR! - <project-root>/storybook-addon-next/examples/node_modules/@storybook/react/bin/index.js
ERR!     at Function.Module._resolveFilename (node:internal/modules/cjs/loader:933:15)
ERR!     at Function.Module._load (node:internal/modules/cjs/loader:778:27)
ERR!     at Module.require (node:internal/modules/cjs/loader:1005:19)
ERR!     at require (node:internal/modules/cjs/helpers:102:18)
ERR!     at getNextjsVersion (<project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/utils.js:9:1)
ERR!     at configureStaticImageImport (<project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/images/webpack.js:14:50)
ERR!     at configureImages (<project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/images/webpack.js:8:5)
ERR!     at <project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/preset.js:29:35
ERR!     at Generator.next (<anonymous>)
ERR!     at fulfilled (<project-root>/storybook-addon-next/examples/node_modules/tslib/tslib.js:114:62) {
ERR!   code: 'MODULE_NOT_FOUND',
ERR!   requireStack: [
ERR!     '<project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/utils.js',
ERR!     '<project-root>/storybook-addon-next/examples/node_modules/storybook-addon-next/dist/preset.js',
ERR!     '<project-root>/storybook-addon-next/examples/node_modules/@storybook/core-common/dist/cjs/presets.js',
ERR!     '<project-root>/storybook-addon-next/examples/node_modules/@storybook/core-common/dist/cjs/index.js',
ERR!     '<project-root>/storybook-addon-next/examples/node_modules/@storybook/core-server/dist/cjs/index.js',
ERR!     '<project-root>/storybook-addon-next/examples/node_modules/@storybook/core/dist/cjs/server.js',
ERR!     '<project-root>/storybook-addon-next/examples/node_modules/@storybook/core/server.js',
ERR!     '<project-root>/storybook-addon-next/examples/node_modules/@storybook/react/dist/cjs/server/index.js',
ERR!     '<project-root>/storybook-addon-next/examples/node_modules/@storybook/react/bin/index.js'
ERR!   ]
ERR! }

WARN Broken build, fix the error above.
WARN You may need to refresh the browser.

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

### Your minimal, reproducible example

https://github.com/volkyeth/storybook-addon-next/tree/broken-with-yarn-workspaces

### Steps to reproduce

```
git clone --branch broken-with-yarn-workspaces git@github.com:volkyeth/storybook-addon-next.git
cd storybook-addon-next/examples
yarn install
cd with-tailwindcss
yarn storybook
```

### Expected behavior

Package should work with yarn workspaces

### How often does this bug happen?

Every time

### Screenshots or Videos

_No response_

### Platform

OS: MacOS

### storybook-addon-next version

1.4.2

### Additional context

I'm not sure what's the fix for this as I'm not very experienced at packaging libs. I'll try to look into it but if someone knows how to make it compatible please advise and I'll try to make a PR

--- a/package.json
+++ b/package.json
@@ -38,7 +38,8 @@
     "test:all": "yarn lint:all && yarn type-check",
     "prepare": "husky install",
     "prepublishOnly": "yarn build",
-    "release": "semantic-release"
+    "release": "semantic-release",
+    "prepack": "yarn build"
   },
   "dependencies": {
     "@storybook/addons": "^6.4.10",

--- a/src/utils.ts
+++ b/src/utils.ts
@@ -5,7 +5,7 @@ import { NextConfig } from 'next'
 
 export const getNextjsVersion = (): string =>
   // eslint-disable-next-line @typescript-eslint/no-var-requires
-  require(path.resolve('node_modules/next/package.json')).version
+  require('next/package.json').version
 
 export const resolveNextConfig = async (
   baseConfig: WebpackConfig
@@ -27,7 +27,7 @@ export const addScopedAlias = (
   if (!baseConfig.resolve.alias) baseConfig.resolve.alias = {}
   const aliasConfig = baseConfig.resolve.alias
 
-  const scopedAlias = path.resolve(`node_modules/${alias ?? name}`)
+  const scopedAlias = require.resolve(`${alias ?? name}`)
   if (Array.isArray(aliasConfig)) {
     aliasConfig.push({
       name,

